guides/asynchronous-tasks/readme.md

```diff
@@ -300,57 +300,59 @@ end
 
 ## Timeouts
 
-You can wrap asynchronous operations in a timeout. This ensures that malicious services don't cause your code to block indefinitely.
+You can wrap asynchronous operations in a timeout. This allow you to put an upper bound on how long the enclosed code will run vs. potentially blocking indefinitely. If the enclosed code hasn't completed by the timeout, it will be interrupted with an {ruby Async::TimeoutError} exception.
```

Comment:
```suggestion
You can wrap asynchronous operations in a timeout. This allows you to put an upper bound on how long the enclosed code will run vs. potentially blocking indefinitely. If the enclosed code hasn't completed by the timeout, it will be interrupted with an {ruby Async::TimeoutError} exception.
```